app/policies/community_policy.rb

```diff
@@ -1,4 +1,25 @@
 class CommunityPolicy < ApplicationPolicy
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      if user 
+        if user.admin
+          scope.all
+        else
+          scope.where("id in (select cu.community_id from community_users cu where cu.user_id = #{@user.id})")
```

Comment:
@luizalbsilva lets use:
scope.where("... cu.user_id = ?", @user.id) to avoid sqli